guides/StreamOutput.rst

```diff
@@ -37,3 +37,78 @@ Stream to S3 Bucket
     $zip->finish();
 
     fclose($zipFile);
+
+Stream to Callback Function
+---------------------------
+
+The CallbackStreamWrapper allows you to stream ZIP data to a custom callback function,
+enabling flexible output handling such as streaming to multiple destinations,
+progress tracking, or data transformation.
+
+.. code-block:: php
+
+    use ZipStream\ZipStream;
+    use ZipStream\Stream\CallbackStreamWrapper;
+
+    // Example 1: Stream to multiple destinations with proper file handling
+    $backupFile = fopen('backup.zip', 'wb');
+    $logFile = fopen('transfer.log', 'ab');
+    
+    $zip = new ZipStream(
+        outputStream: CallbackStreamWrapper::open(function (string $data) use ($backupFile, $logFile) {
+            // Send to browser
+            echo $data;
+            
+            // Save to file efficiently
+            fwrite($backupFile, $data);
+            
+            // Log transfer progress
+            fwrite($logFile, "Transferred " . strlen($data) . " bytes\n");
+        }),
+        sendHttpHeaders: false,
+    );
+
+    $zip->addFile('hello.txt', 'Hello World!');
+    $zip->finish();
+    
+    // Clean up resources
+    fclose($backupFile);
+    fclose($logFile);
+
+.. code-block:: php
+
+    // Example 2: Progress tracking
+    $totalBytes = 0;
+    $zip = new ZipStream(
+        outputStream: CallbackStreamWrapper::open(function (string $data) use (&$totalBytes) {
+            $totalBytes += strlen($data);
+            echo "Progress: {$totalBytes} bytes written\n";
```

Comment:
I don't like this example since it outputs both data and status messages to the browser which will result in an invalid zip. Can we replace this with some dummy code like `reportProgress($totalBytes);` or something like that?